conda_build/skeletons/cran.py

```diff
@@ -1423,13 +1423,20 @@ def get_license_file(license_text):
     because they would need to be combined with the license information provided
     by the package authors (e.g. copyright owners and date).
 
-    This function returns the path to the license file for the unambiguous
-    cases. Any time an R package refers to one of the templates or a custom
-    license file (e.g. 'GPL-2 | file LICENSE'), an empty string is returned.
+    This function returns the path to a R base provided complete license file
+    by matching license_text either unambiguous or by matching optional '|'
+    seperated licenses in the order defined by d_license, e.g.:
+      * Single unambiguous complete license: 'GPL-2' -> 'GPL-2'
+      * Optional complete and custom license: 'GPL-2 | file LICENSE' -> 'GPL-2'
+      * Two optional complete licenses: 'GPL-2 | GPL-3' -> 'GPL-2'
+      * Two optional complete licenses: 'GPL-3 | GPL-2' -> 'GPL-2'
```

Comment:
Is favoring `GPL-2` in this final case a technical or legal reason? In other words, why is this behavior chosen over these two alternatives:

1. Always choose the first license that the package author lists
1. Always favor GPL-3 over GPL-2 since it is the latest version of the GPL

Reply:
I thought there are two options for the match order:
1. The order of the licenses in `license_text`: In this case the package upstream can influence it.
2. The order of the sorted keys of `d_license`: In this case the cran skeleton defines the order (which could be also legal driven)

I just did 2, because thats what I just started with :) And I also like that both `GPL-2 | GPL-3` (now new) and `GPL (>= 2)` (was already the case before) result in GPL2, even though both could match to GPL3... just for a kind of consistency... not because I prefer the one over the other... 

I am also happy to change the implementation to approach 1. And of course adding tests makes sense.

Most important thing is that the matching is deterministic and resulting in a correct license.

(Best thing would be maybe to include multiple license files, that would also allow to add combined licenses and the match oder would not matter anymore, but that requires probably a change of the meta.yaml schema to allow `license_file` being a list.. sounds like a bigger exercise)

Reply:
>    The order of the licenses in license_text: In this case the package upstream can influence it.

I don't follow. How can the upstream package influence it? If the authors put a given license as the first choice, I think it is safe to assume that we can use it.

>    The order of the sorted keys of d_license: In this case the cran skeleton defines the order (which could be also legal driven)

This seems arbitrary since it is alphabetical.

> I am also happy to change the implementation to approach 1. And of course adding tests makes sense.

To test `GPL-3 | GPL-2`, you can use `r-magree`. It's actually the only package I could find with that exact license.

> Most important thing is that the matching is deterministic and resulting in a correct license.

Agree 100%. The current problem is that my `for` loop solution worked well for the single license case, but isn't very sophisticated. To handle multiple licenses, it may be necessary to rethink the approach.

> (Best thing would be maybe to include multiple license files, that would also allow to add combined licenses and the match order would not matter anymore, but that requires probably a change of the meta.yaml schema to allow license_file being a list.. sounds like a bigger exercise)

Agreed. I had investigated this possibility. Since it is only possible to include one file, I was thinking long-term we could concatenate files when necessary (e.g. for the common case of the MIT license, concatenate the file `LICENSE` on top of the MIT template file). But since that would require reading and writing from disk, I didn't attempt that in my initial PR.

Reply:
ok. Found some time to change the matching to respect the given license and extended the tests.

Reply:
Closing as this was now done in a different way.